tests/phases/design/test_semisubmersible_design.py

```diff
@@ -65,3 +69,141 @@ def test_design_kwargs():
         cost = s.total_cost
 
         assert cost != base_cost
+
+
+config_custom = extract_library_specs(
+    "config", "semisubmersible_design_custom"
+)
+
+
+def test_calc_geometric_scale_factor():
+
+    config = deepcopy(config_custom)
+
+    custom = CustomSemiSubmersibleDesign(config)
+    custom.run()
+
+    assert custom.geom_scale_factor == 1.0
+
+    config["turbine"]["rotor_diameter"] = 220
+    custom.run()
+
+    assert custom.geom_scale_factor == pytest.approx(0.9392, 1e-4)
+
+    with pytest.raises(KeyError):
+        config["turbine"]["rotor_diameter"] = None
+        custom.run()
+
+
+def test_bouyant_column_volume():
+
+    custom = CustomSemiSubmersibleDesign(config_custom)
+    custom.run()
+
+    assert custom.bouyant_column_volume == pytest.approx(72.5136, 1e-4)
+
+
+def test_center_column_volume():
+
+    custom = CustomSemiSubmersibleDesign(config_custom)
+    custom.run()
+
+    assert custom.center_column_volume == pytest.approx(39.4059, 1e-4)
+
+
+def test_pontoon_volume():
+
+    custom = CustomSemiSubmersibleDesign(config_custom)
+    custom.run()
+
+    assert custom.pontoon_volume == pytest.approx(90.4021, 1e-4)
+
+
+def test_strut_volume():
+
+    custom = CustomSemiSubmersibleDesign(config_custom)
+    custom.run()
+
+    assert custom.strut_volume == pytest.approx(32.9219, 1e-4)
+
+
+def test_substructure_steel_mass():
+
+    custom = CustomSemiSubmersibleDesign(config_custom)
+    custom.run()
+
+    assert custom.substructure_steel_mass == pytest.approx(5000, 1e0)
+
+
+def test_ballast_mass():
+
+    custom = CustomSemiSubmersibleDesign(config_custom)
+    custom.run()
+
+    assert custom.ballast_mass == 2540
+
+
+def test_tower_interface_mass():
+
+    custom = CustomSemiSubmersibleDesign(config_custom)
+    custom.run()
+
+    assert custom.tower_interface_mass == 100
+
+
+def test_substructure_steel_cost():
+
+    custom = CustomSemiSubmersibleDesign(config_custom)
+    custom.run()
+
+    assert custom.steel_cr == 4500
+
+
+def test_substructure_mass():
+
+    custom = CustomSemiSubmersibleDesign(config_custom)
+    custom.run()
+
+    assert custom.substructure_mass == pytest.approx(7642, 1e0)
+
+
+def test_substructure_unit_cost():
+
+    custom = CustomSemiSubmersibleDesign(config_custom)
+    custom.run()
+
+    assert custom.substructure_unit_cost == pytest.approx(2.3343e7, 1e-4)
+
+
+def test_custom_design_kwargs():
```

Comment:
I'm realizing that this was the design pattern for the original set of tests for the semi submersible design, but I was assuming this would also check that the input parameters are being fed through correctly. Would you be able to add basic checks for the new and existing model?